### PR TITLE
Add CSS parser and average ages tasks

### PR DESCRIPTION
**Pull request by dreyhiflden**

### CODE REVIEW

File: css_parsing.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  width: auto;
+`;
+
+/**
+ * Функция принимает строку со стилями
+ * и возвращает объект у которого
+ * названия CSS свойств это ключи
+ * значения - строки
+ */
+function convertToObject(sourceString) {
```
Comment (carrot-e) [inline css_parsing.js]: Good!
One minor improvement idea: 
you don't need the `resultObject` variable at all. Instead you should use `reduce` to create the expected object.

File: average_ages.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+];
+
+// Существует набор данных на основе него
+
+// подсчитайте средний возраст мужчин
+// подсчитайте средний возраст женщин
+// подсчитайте среднюю разницу в возрасте между матерями и их детьми
+// выполните выше указанные подсчеты для каждого из столетий Назначаем столетию людей, беря их год смерти, деля его на 100 и округляя: Math.ceil(person.died / 100).
+
+function getAverageMenAge(people) {
```
Comment (carrot-e) [inline average_ages.js]: `getAverageMenAge` and `getAverageWomenAge` functions are very similar. Can you think how you can use the same code for getting women's and men's avg ages?

File: average_ages.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  return people
+    .filter(i => i['sex'] === 'm')
+    .map(i => i['died'] - i['born'])
+    .reduce((acc, cur, index, arr) => {
+      let getSum = acc + cur;
+      return (index === arr.length-1) ? getSum / arr.length : getSum;
```
Comment (carrot-e) [inline average_ages.js]: Codestyle: `arr.length - 1`.

File: average_ages.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+// подсчитайте средний возраст женщин
+// подсчитайте среднюю разницу в возрасте между матерями и их детьми
+// выполните выше указанные подсчеты для каждого из столетий Назначаем столетию людей, беря их год смерти, деля его на 100 и округляя: Math.ceil(person.died / 100).
+
+function getAverageMenAge(people) {
+  return people
+    .filter(i => i['sex'] === 'm')
+    .map(i => i['died'] - i['born'])
+    .reduce((acc, cur, index, arr) => {
+      let getSum = acc + cur;
```
Comment (carrot-e) [inline average_ages.js]: `getSum` is a good name for a function, not for a variable.

File: average_ages.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    });
+}
+
+function getAverageDiff(people) {
+  let arrWithMothers = people.filter(i => i['mother'] !== null);
+  let arrDifferences = [];
+
+  arrWithMothers
+    .map((item, index, array) => {
+    let mother = arrWithMothers[index]['mother'];
```
Comment (carrot-e) [inline average_ages.js]: Codestyle: add a nesting level

Comment (carrot-e) [reply average_ages.js]: `arrWithMothers[index]` === `item`?

File: average_ages.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+    array
+      .map((obj) => {
```
Comment (carrot-e) [inline average_ages.js]: I think, `.forEach` or even `.find` would be more suitable here.

File: average_ages.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+function getAverageDiff(people) {
+  let arrWithMothers = people.filter(i => i['mother'] !== null);
+  let arrDifferences = [];
+
+  arrWithMothers
+    .map((item, index, array) => {
+    let mother = arrWithMothers[index]['mother'];
+
+    array
```
Comment (carrot-e) [inline average_ages.js]: You are not taking into account those women, whose `mother` is null.
They can be someones' mothers nevertheless.

For example, `Joanna de Causmaecker` has `"mother":null`, but she is a mother of `Maria van Brussel`.

File: average_ages.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+  function getCentury(arrayItem) {
+    return Math.ceil(arrayItem['died'] / 100);
+  }
+
+  function pushToObject(arrayItem) {
+    return stats[getCentury(arrayItem)] = {
+    };
+  }
+
+  for(let key in stats) {
```
Comment (carrot-e) [inline average_ages.js]: Codestyle: `for (let`

File: css_parsing.js (hunk 1 above)
Comment (dreyhiflden) [reply css_parsing.js]: Thanks, changed

**Pull request merged**